packages/@react-spectrum/listbox/src/ListBoxOption.tsx

```diff
@@ -74,20 +74,24 @@ export function ListBoxOption<T>(props: OptionProps<T>) {
     ? <Text>{rendered}</Text>
     : rendered;
 
+  let isKeyboardModality = isFocusVisible();
+
   return (
     <FocusRing focusRingClass={classNames(styles, 'focus-ring')}>
       <div
-        {...mergeProps(optionProps, shouldFocusOnHover ? {} : hoverProps)}
+        {...mergeProps(optionProps, shouldFocusOnHover ? hoverProps : {})}
```

Comment:
`isHovered` wasn't updating without this flip, think this is what it was supposed to be